### ISSUE TITLE
Doesn't show octomap in Perception pipeline tutorial 

### ISSUE THREAD

**Issue by huiwenzhang** [ros-planning/moveit_tutorials#505]
### Description

I followed instructions in http://docs.ros.org/melodic/api/moveit_tutorials/html/doc/perception_pipeline/perception_pipeline_tutorial.html. The launch file should spawn the rviz and  a franka arm and the octomap show. However, only arm is correctly displayed, no octomap.

### Your environment
* ROS Distro: [Melodic]
* OS Version: Ubuntu 18.04
* Source build with master branch

### Steps to reproduce
To reproduce, just follow instructions in http://docs.ros.org/melodic/api/moveit_tutorials/html/doc/perception_pipeline/perception_pipeline_tutorial.html.

### Expected behaviour
The franka arm shows in rviz with an Octomap represents the obstacles in the environment.

### Backtrace or Console output

When I run `roslaunch moveit_tutorials obstacle_avoidance_demo.launch`, the console outputs
```
started roslaunch server http://alvin-X280:34129/
SUMMARY
========
PARAMETERS
 * /joint_state_publisher/use_gui: False
 * /move_group/allow_trajectory_execution: True
 * /move_group/controller_list: [{'joints': ['pan...
 * /move_group/hand/planner_configs: ['SBLkConfigDefau...
 * /move_group/initial: [{'pose': 'ready'...
 * /move_group/jiggle_fraction: 0.05
 * /move_group/max_range: 5.0
 * /move_group/max_safe_path_cost: 1
 * /move_group/moveit_controller_manager: moveit_fake_contr...
 * /move_group/moveit_manage_controllers: True
 * /move_group/octomap_frame: camera_rgb_optica...
 * /move_group/octomap_resolution: 0.25
 * /move_group/panda_arm/planner_configs: ['SBLkConfigDefau...
 * /move_group/panda_arm_hand/planner_configs: ['SBLkConfigDefau...
 * /move_group/planner_configs/BFMTkConfigDefault/balanced: 0
 * /move_group/planner_configs/BFMTkConfigDefault/cache_cc: 1
 * /move_group/planner_configs/BFMTkConfigDefault/extended_fmt: 1
 * /move_group/planner_configs/BFMTkConfigDefault/heuristics: 1
 * /move_group/planner_configs/BFMTkConfigDefault/nearest_k: 1
 * /move_group/planner_configs/BFMTkConfigDefault/num_samples: 1000
 * /move_group/planner_configs/BFMTkConfigDefault/optimality: 1
 * /move_group/planner_configs/BFMTkConfigDefault/radius_multiplier: 1.0
 * /move_group/planner_configs/BFMTkConfigDefault/type: geometric::BFMT
 * /move_group/planner_configs/BKPIECEkConfigDefault/border_fraction: 0.9
 * /move_group/planner_configs/BKPIECEkConfigDefault/failed_expansion_score_factor: 0.5
 * /move_group/planner_configs/BKPIECEkConfigDefault/min_valid_path_fraction: 0.5
 * /move_group/planner_configs/BKPIECEkConfigDefault/range: 0.0
 * /move_group/planner_configs/BKPIECEkConfigDefault/type: geometric::BKPIECE
 * /move_group/planner_configs/BiESTkConfigDefault/range: 0.0
 * /move_group/planner_configs/BiESTkConfigDefault/type: geometric::BiEST
 * /move_group/planner_configs/BiTRRTkConfigDefault/cost_threshold: 1e300
 * /move_group/planner_configs/BiTRRTkConfigDefault/frountier_node_ratio: 0.1
 * /move_group/planner_configs/BiTRRTkConfigDefault/frountier_threshold: 0.0
 * /move_group/planner_configs/BiTRRTkConfigDefault/init_temperature: 100
 * /move_group/planner_configs/BiTRRTkConfigDefault/range: 0.0
 * /move_group/planner_configs/BiTRRTkConfigDefault/temp_change_factor: 0.1
 * /move_group/planner_configs/BiTRRTkConfigDefault/type: geometric::BiTRRT
 * /move_group/planner_configs/ESTkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/ESTkConfigDefault/range: 0.0
 * /move_group/planner_configs/ESTkConfigDefault/type: geometric::EST
 * /move_group/planner_configs/FMTkConfigDefault/cache_cc: 1
 * /move_group/planner_configs/FMTkConfigDefault/extended_fmt: 1
 * /move_group/planner_configs/FMTkConfigDefault/heuristics: 0
 * /move_group/planner_configs/FMTkConfigDefault/nearest_k: 1
 * /move_group/planner_configs/FMTkConfigDefault/num_samples: 1000
 * /move_group/planner_configs/FMTkConfigDefault/radius_multiplier: 1.1
 * /move_group/planner_configs/FMTkConfigDefault/type: geometric::FMT
 * /move_group/planner_configs/KPIECEkConfigDefault/border_fraction: 0.9
 * /move_group/planner_configs/KPIECEkConfigDefault/failed_expansion_score_factor: 0.5
 * /move_group/planner_configs/KPIECEkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/KPIECEkConfigDefault/min_valid_path_fraction: 0.5
 * /move_group/planner_configs/KPIECEkConfigDefault/range: 0.0
 * /move_group/planner_configs/KPIECEkConfigDefault/type: geometric::KPIECE
 * /move_group/planner_configs/LBKPIECEkConfigDefault/border_fraction: 0.9
 * /move_group/planner_configs/LBKPIECEkConfigDefault/min_valid_path_fraction: 0.5
 * /move_group/planner_configs/LBKPIECEkConfigDefault/range: 0.0
 * /move_group/planner_configs/LBKPIECEkConfigDefault/type: geometric::LBKPIECE
 * /move_group/planner_configs/LBTRRTkConfigDefault/epsilon: 0.4
 * /move_group/planner_configs/LBTRRTkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/LBTRRTkConfigDefault/range: 0.0
 * /move_group/planner_configs/LBTRRTkConfigDefault/type: geometric::LBTRRT
 * /move_group/planner_configs/LazyPRMkConfigDefault/range: 0.0
 * /move_group/planner_configs/LazyPRMkConfigDefault/type: geometric::LazyPRM
 * /move_group/planner_configs/LazyPRMstarkConfigDefault/type: geometric::LazyPR...
 * /move_group/planner_configs/PDSTkConfigDefault/type: geometric::PDST
 * /move_group/planner_configs/PRMkConfigDefault/max_nearest_neighbors: 10
 * /move_group/planner_configs/PRMkConfigDefault/type: geometric::PRM
 * /move_group/planner_configs/PRMstarkConfigDefault/type: geometric::PRMstar
 * /move_group/planner_configs/ProjESTkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/ProjESTkConfigDefault/range: 0.0
 * /move_group/planner_configs/ProjESTkConfigDefault/type: geometric::ProjEST
 * /move_group/planner_configs/RRTConnectkConfigDefault/range: 0.0
 * /move_group/planner_configs/RRTConnectkConfigDefault/type: geometric::RRTCon...
 * /move_group/planner_configs/RRTkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/RRTkConfigDefault/range: 0.0
 * /move_group/planner_configs/RRTkConfigDefault/type: geometric::RRT
 * /move_group/planner_configs/RRTstarkConfigDefault/delay_collision_checking: 1
 * /move_group/planner_configs/RRTstarkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/RRTstarkConfigDefault/range: 0.0
 * /move_group/planner_configs/RRTstarkConfigDefault/type: geometric::RRTstar
 * /move_group/planner_configs/SBLkConfigDefault/range: 0.0
 * /move_group/planner_configs/SBLkConfigDefault/type: geometric::SBL
 * /move_group/planner_configs/SPARSkConfigDefault/dense_delta_fraction: 0.001
 * /move_group/planner_configs/SPARSkConfigDefault/max_failures: 1000
 * /move_group/planner_configs/SPARSkConfigDefault/sparse_delta_fraction: 0.25
 * /move_group/planner_configs/SPARSkConfigDefault/stretch_factor: 3.0
 * /move_group/planner_configs/SPARSkConfigDefault/type: geometric::SPARS
 * /move_group/planner_configs/SPARStwokConfigDefault/dense_delta_fraction: 0.001
 * /move_group/planner_configs/SPARStwokConfigDefault/max_failures: 5000
 * /move_group/planner_configs/SPARStwokConfigDefault/sparse_delta_fraction: 0.25
 * /move_group/planner_configs/SPARStwokConfigDefault/stretch_factor: 3.0
 * /move_group/planner_configs/SPARStwokConfigDefault/type: geometric::SPARStwo
 * /move_group/planner_configs/STRIDEkConfigDefault/degree: 16
 * /move_group/planner_configs/STRIDEkConfigDefault/estimated_dimension: 0.0
 * /move_group/planner_configs/STRIDEkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/STRIDEkConfigDefault/max_degree: 18
 * /move_group/planner_configs/STRIDEkConfigDefault/max_pts_per_leaf: 6
 * /move_group/planner_configs/STRIDEkConfigDefault/min_degree: 12
 * /move_group/planner_configs/STRIDEkConfigDefault/min_valid_path_fraction: 0.2
 * /move_group/planner_configs/STRIDEkConfigDefault/range: 0.0
 * /move_group/planner_configs/STRIDEkConfigDefault/type: geometric::STRIDE
 * /move_group/planner_configs/STRIDEkConfigDefault/use_projected_distance: 0
 * /move_group/planner_configs/TRRTkConfigDefault/frountierNodeRatio: 0.1
 * /move_group/planner_configs/TRRTkConfigDefault/frountier_threshold: 0.0
 * /move_group/planner_configs/TRRTkConfigDefault/goal_bias: 0.05
 * /move_group/planner_configs/TRRTkConfigDefault/init_temperature: 10e-6
 * /move_group/planner_configs/TRRTkConfigDefault/k_constant: 0.0
 * /move_group/planner_configs/TRRTkConfigDefault/max_states_failed: 10
 * /move_group/planner_configs/TRRTkConfigDefault/min_temperature: 10e-10
 * /move_group/planner_configs/TRRTkConfigDefault/range: 0.0
 * /move_group/planner_configs/TRRTkConfigDefault/temp_change_factor: 2.0
 * /move_group/planner_configs/TRRTkConfigDefault/type: geometric::TRRT
 * /move_group/planner_configs/TrajOptDefault/type: geometric::TrajOpt
 * /move_group/planning_plugin: ompl_interface/OM...
 * /move_group/planning_scene_monitor/publish_geometry_updates: True
 * /move_group/planning_scene_monitor/publish_planning_scene: True
 * /move_group/planning_scene_monitor/publish_state_updates: True
 * /move_group/planning_scene_monitor/publish_transforms_updates: True
 * /move_group/request_adapters: default_planner_r...
 * /move_group/sensors: [{'point_subsampl...
 * /move_group/start_state_max_bounds_error: 0.1
 * /move_group/trajectory_execution/allowed_execution_duration_scaling: 1.2
 * /move_group/trajectory_execution/allowed_goal_duration_margin: 0.5
 * /move_group/trajectory_execution/allowed_start_tolerance: 0.01
 * /robot_description: <?xml version="1....
 * /robot_description_kinematics/panda_arm/kinematics_solver: kdl_kinematics_pl...
 * /robot_description_kinematics/panda_arm/kinematics_solver_search_resolution: 0.005
 * /robot_description_kinematics/panda_arm/kinematics_solver_timeout: 0.05
 * /robot_description_planning/joint_limits/panda_finger_joint1/has_acceleration_limits: False
 * /robot_description_planning/joint_limits/panda_finger_joint1/has_velocity_limits: True
 * /robot_description_planning/joint_limits/panda_finger_joint1/max_acceleration: 0
 * /robot_description_planning/joint_limits/panda_finger_joint1/max_velocity: 0.1
 * /robot_description_planning/joint_limits/panda_finger_joint2/has_acceleration_limits: False
 * /robot_description_planning/joint_limits/panda_finger_joint2/has_velocity_limits: True
 * /robot_description_planning/joint_limits/panda_finger_joint2/max_acceleration: 0
 * /robot_description_planning/joint_limits/panda_finger_joint2/max_velocity: 0.1
 * /robot_description_planning/joint_limits/panda_joint1/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/panda_joint1/has_velocity_limits: True
 * /robot_description_planning/joint_limits/panda_joint1/max_acceleration: 3.75
 * /robot_description_planning/joint_limits/panda_joint1/max_velocity: 2.175
 * /robot_description_planning/joint_limits/panda_joint2/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/panda_joint2/has_velocity_limits: True
 * /robot_description_planning/joint_limits/panda_joint2/max_acceleration: 1.875
 * /robot_description_planning/joint_limits/panda_joint2/max_velocity: 2.175
 * /robot_description_planning/joint_limits/panda_joint3/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/panda_joint3/has_velocity_limits: True
 * /robot_description_planning/joint_limits/panda_joint3/max_acceleration: 2.5
 * /robot_description_planning/joint_limits/panda_joint3/max_velocity: 2.175
 * /robot_description_planning/joint_limits/panda_joint4/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/panda_joint4/has_velocity_limits: True
 * /robot_description_planning/joint_limits/panda_joint4/max_acceleration: 3.125
 * /robot_description_planning/joint_limits/panda_joint4/max_velocity: 2.175
 * /robot_description_planning/joint_limits/panda_joint5/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/panda_joint5/has_velocity_limits: True
 * /robot_description_planning/joint_limits/panda_joint5/max_acceleration: 3.75
 * /robot_description_planning/joint_limits/panda_joint5/max_velocity: 2.61
 * /robot_description_planning/joint_limits/panda_joint6/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/panda_joint6/has_velocity_limits: True
 * /robot_description_planning/joint_limits/panda_joint6/max_acceleration: 5
 * /robot_description_planning/joint_limits/panda_joint6/max_velocity: 2.61
 * /robot_description_planning/joint_limits/panda_joint7/has_acceleration_limits: True
 * /robot_description_planning/joint_limits/panda_joint7/has_velocity_limits: True
 * /robot_description_planning/joint_limits/panda_joint7/max_acceleration: 5
 * /robot_description_planning/joint_limits/panda_joint7/max_velocity: 2.61
 * /robot_description_semantic: <?xml version="1....
 * /rosdistro: melodic
 * /rosversion: 1.14.5
 * /rviz_alvin_X280_7493_7888606938195610446/panda_arm/kinematics_solver: kdl_kinematics_pl...
 * /rviz_alvin_X280_7493_7888606938195610446/panda_arm/kinematics_solver_search_resolution: 0.005
 * /rviz_alvin_X280_7493_7888606938195610446/panda_arm/kinematics_solver_timeout: 0.05
 * /source_list: ['/move_group/fak...

NODES
  /
    joint_state_desired_publisher (topic_tools/relay)
    joint_state_publisher (joint_state_publisher/joint_state_publisher)
    move_group (moveit_ros_move_group/move_group)
    point_clouds (moveit_tutorials/bag_publisher_maintain_time)
    robot_state_publisher (robot_state_publisher/robot_state_publisher)
    rviz_alvin_X280_7493_7888606938195610446 (rviz/rviz)
    to_panda_base (tf2_ros/static_transform_publisher)
    to_temp_link (tf2_ros/static_transform_publisher)
    virtual_joint_broadcaster_1 (tf2_ros/static_transform_publisher)

auto-starting new master
process[master]: started with pid [7512]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 7df4aa88-badb-11ea-adb0-3c6aa714c58f
process[rosout-1]: started with pid [7524]
started core service [/rosout]
process[virtual_joint_broadcaster_1-2]: started with pid [7531]
process[joint_state_publisher-3]: started with pid [7532]
process[joint_state_desired_publisher-4]: started with pid [7533]
process[robot_state_publisher-5]: started with pid [7539]
process[move_group-6]: started with pid [7544]
process[rviz_alvin_X280_7493_7888606938195610446-7]: started with pid [7550]
process[point_clouds-8]: started with pid [7554]
process[to_temp_link-9]: started with pid [7555]
process[to_panda_base-10]: started with pid [7556]
[ INFO] [1593526247.994313073]: Loading robot model 'panda'...
[ INFO] [1593526248.072384688]: rviz version 1.13.9
[ INFO] [1593526248.072437660]: compiled against Qt version 5.9.5
[ INFO] [1593526248.072455069]: compiled against OGRE version 1.9.0 (Ghadamon)
[ INFO] [1593526248.083070289]: Forcing OpenGl version 0.
[ INFO] [1593526248.092734614]: Publishing maintained planning scene on 'monitored_planning_scene'
[ INFO] [1593526248.095097266]: MoveGroup debug mode is ON
Starting planning scene monitors...
[ INFO] [1593526248.095139232]: Starting planning scene monitor
[ INFO] [1593526248.096804911]: Listening to '/planning_scene'
[ INFO] [1593526248.096835033]: Starting world geometry update monitor for collision objects, attached objects, octomap updates.
[ INFO] [1593526248.098550871]: Listening to '/collision_object'
[ INFO] [1593526248.100507261]: Listening to '/planning_scene_world' for planning scene world geometry
[ INFO] [1593526248.120085867]: Listening to '/camera/depth_registered/points' using message filter with target frame 'world '
[ INFO] [1593526248.123451528]: Listening to '/attached_collision_object' for attached collision objects
Planning scene monitors started.
[ INFO] [1593526248.195409673]: Using planning interface 'OMPL'
[ INFO] [1593526248.198474608]: Param 'default_workspace_bounds' was not set. Using default value: 10
[ INFO] [1593526248.198864031]: Param 'start_state_max_bounds_error' was set to 0.1
[ INFO] [1593526248.199204597]: Param 'start_state_max_dt' was not set. Using default value: 0.5
[ INFO] [1593526248.199604766]: Param 'start_state_max_dt' was not set. Using default value: 0.5
[ INFO] [1593526248.199997148]: Param 'jiggle_fraction' was set to 0.05
[ INFO] [1593526248.200391819]: Param 'max_sampling_attempts' was not set. Using default value: 100
[ INFO] [1593526248.200476293]: Using planning request adapter 'Fix Workspace Bounds'
[ INFO] [1593526248.200502008]: Using planning request adapter 'Fix Start State Bounds'
[ INFO] [1593526248.200525532]: Using planning request adapter 'Fix Start State In Collision'
[ INFO] [1593526248.200551074]: Using planning request adapter 'Fix Start State Path Constraints'
[ INFO] [1593526248.200573097]: Using planning request adapter 'Resolve constraint frames to robot links'
[ INFO] [1593526248.200590600]: Using planning request adapter 'Add Time Parameterization'
[ INFO] [1593526248.207914185]: Loading robot model 'panda'...
[ WARN] [1593526248.209999569]: Unable to update multi-DOF joint 'virtual_joint': Failure to lookup transform between 'world' and 'panda_link0' with TF exception: Could not find a connection between 'world' and 'panda_link0' because they are not part of the same tree.Tf has two or more unconnected trees.
[ INFO] [1593526248.245320146]: Set joints of group 'panda_arm' to pose 'ready'.
[ INFO] [1593526248.245358277]: Set joints of group 'hand' to pose 'open'.
[ INFO] [1593526248.245603021]: Fake controller 'fake_panda_arm_controller' with joints [ panda_joint1 panda_joint2 panda_joint3 panda_joint4 panda_joint5 panda_joint6 panda_joint7 ]
[ INFO] [1593526248.245928336]: Fake controller 'fake_hand_controller' with joints [ panda_finger_joint1 panda_finger_joint2 ]
[ INFO] [1593526248.246291460]: Returned 2 controllers in list
[ INFO] [1593526248.254213228]: Trajectory execution is managing controllers
Loading 'move_group/ApplyPlanningSceneService'...
Loading 'move_group/ClearOctomapService'...
Loading 'move_group/MoveGroupCartesianPathService'...
Loading 'move_group/MoveGroupExecuteTrajectoryAction'...
Loading 'move_group/MoveGroupGetPlanningSceneService'...
Loading 'move_group/MoveGroupKinematicsService'...
Loading 'move_group/MoveGroupMoveAction'...
Loading 'move_group/MoveGroupPickPlaceAction'...
Loading 'move_group/MoveGroupPlanService'...
Loading 'move_group/MoveGroupQueryPlannersService'...
Loading 'move_group/MoveGroupStateValidationService'...
[ INFO] [1593526248.305197566]: 

********************************************************
* MoveGroup using: 
*     - ApplyPlanningSceneService
*     - ClearOctomapService
*     - CartesianPathService
*     - ExecuteTrajectoryAction
*     - GetPlanningSceneService
*     - KinematicsService
*     - MoveAction
*     - PickPlaceAction
*     - MotionPlanService
*     - QueryPlannersService
*     - StateValidationService
********************************************************

[ INFO] [1593526248.305266119]: MoveGroup context using planning plugin ompl_interface/OMPLPlanner
[ INFO] [1593526248.305289798]: MoveGroup context initialization complete

You can start planning now!

[ INFO] [1593526248.524273836]: Stereo is NOT SUPPORTED
[ INFO] [1593526248.524355554]: OpenGl version: 3 (GLSL 1.3).
[ INFO] [1593526248.693679293]: Loading robot model 'panda'...
[ INFO] [1593526249.072796106]: Loading robot model 'panda'...
[ INFO] [1593526249.156872495]: Starting planning scene monitor
[ INFO] [1593526249.159388537]: Listening to '/move_group/monitored_planning_scene'
QObject::connect: Cannot queue arguments of type 'QVector<int>'
(Make sure 'QVector<int>' is registered using qRegisterMetaType().)
QObject::connect: Cannot queue arguments of type 'QVector<int>'
(Make sure 'QVector<int>' is registered using qRegisterMetaType().)
```
I checked out the outputs, seems nothing going wrong here, but the rviz doesn't show octomap. I echoed the `/move_group/monitored_planning_scene` topic, nothing shows.

**Comment by v4hn** [comment]
Did you run into the mesa bug [we only documented recently in master](https://ros-planning.github.io/moveit_tutorials/doc/perception_pipeline/perception_pipeline_tutorial.html#running-the-interface)?

**Comment by huiwenzhang** [comment]
> Did you run into the mesa bug [we only documented recently in master](https://ros-planning.github.io/moveit_tutorials/doc/perception_pipeline/perception_pipeline_tutorial.html#running-the-interface)?

I set the environment variable as suggested by the link, but it doesn't work. I don't think this is caused by the rendering problem, since the arm is visualized correctly. The strange point is the topic `/move_group/monitored_planning_scene` seems to be empty, while  the pointcloud msg is published normally.

**Comment by lianghongzhuo** [comment]
I have the save issues in moveit melodic version.  Only see Franka robot, no octomap.
Any updates on this issues?

**Comment by lianghongzhuo** [comment]
`rostopic echo /camera/depth_registered/points` can output point msgs
but can not see the pointcloud using rviz

**Comment by lianghongzhuo** [comment]
I have fixed the issue, The launch file is wrong. I will open a pull request to fix the bug.

**Comment by rhaschke** [comment]
Fixed via backporting #491.

**Comment by lianghongzhuo** [comment]
Thanks for the fix.
